### PR TITLE
Feature/delete buttons

### PR DESCRIPTION
**Pull request by ddelpiano**
- delete button feature implemented for netParams
- dialog box implemented on delete action to confirm or cancel the action

### CODE REVIEW

**Comment by ddelpiano** [comment]
@tarelli @adrianq this is linked to these other PRs:
- https://github.com/MetaCell/NetPyNE-UI/pull/33
- https://github.com/openworm/org.geppetto.frontend/pull/809

File: components/general/NetPyNEAddNew.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import React, { Component } from 'react';
 import ContentAdd from 'material-ui/svg-icons/content/add';
 import FloatingActionButton from 'material-ui/FloatingActionButton';
+require('./NetPyNEAddNew.less');
```
Comment (adrianq) [inline components/general/NetPyNEAddNew.js]: I haven't started testing it properly but sth I have just noticed when compiling: this file is missing, probably you forgot to push it :)

Comment (ddelpiano) [reply components/general/NetPyNEAddNew.js]: @adrianq thx for the spot, fixed, I was originally planning to use this but I did not need at the end.

**Comment by adrianq** [comment]
@ddelpiano My alternative proposal can be found here -> https://github.com/MetaCell/geppetto-netpyne/pull/37

**Comment by adrianq** [comment]
@ddelpiano ready to be reviewed?

**Comment by ddelpiano** [comment]
@adrianq now is ready, there is a small thing that I am still fixing but it's pure styling, something related to the icon for the Cell Rule->Section->Mechanism when you go hover this and the bin should appear.
The logic will be untouched.
Thx

**Comment by ddelpiano** [comment]
@adrianq @tarelli quick one for both of you before to go further on this.
As you can see I completed the delete feature also for the nested components of CellRule (section and mechanism).
I got a cosmetic problem with mechanism, since this is using FontIcon I cannot fill this on hover with the font icon I used for the other.

Should I change this to use the Floating Action button?
Any other alternative?
At the moment when on hover the gear icon disappear and the bin appear, pretty sure you both are not gonna like.
Thx!

**Comment by adrianq** [comment]
@ddelpiano I have just implemented something that fixes the thumbnail issue. Just a bit of css and html but quite a bit of pain ;) You can see it here https://github.com/MetaCell/geppetto-netpyne/pull/32/commits/c3b000b096516e9f360093ead580ffe347a849dc Only thing missing is the hover overlayer that increases the thumbnail opacity

I also fixed another issue: if you pressed cancel in the delete dialog and tried to open it again in the same thumbnail it was not working properly and the dialog didn't show up.

Thanks for your patience with this PR but I think it is looking very good now!

**Pull request merged**